docs/src/man/reshaping_and_pivoting.md

```diff
@@ -72,14 +276,181 @@ This repeats the original columns N times where N is the number of columns stack
 
 For more details on the storage representation, see:
 
-```julia
-dump(stackdf(iris))
+```jldoctest reshape
+julia> dump(stackdf(iris))
+DataFrames.DataFrame  600 observations of 4 variables
+  variable: DataFrames.RepeatedVector{Symbol}
+    parent: Array{Symbol}((4,))
+      1: Symbol SepalLength
+      2: Symbol SepalWidth
+      3: Symbol PetalLength
+      4: Symbol PetalWidth
+    inner: Int64 150
+    outer: Int64 1  value: DataFrames.StackedVector
+    components: Array{Any}((4,))
+      1: Array{Float64}((150,)) [5.1, 4.9, 4.7, 4.6, 5.0, 5.4, 4.6, 5.0, 4.4, 4.9  …  6.7, 6.9, 5.8, 6.8, 6.7, 6.7, 6.3, 6.5, 6.2, 5.9]
+      2: Array{Float64}((150,)) [3.5, 3.0, 3.2, 3.1, 3.6, 3.9, 3.4, 3.4, 2.9, 3.1  …  3.1, 3.1, 2.7, 3.2, 3.3, 3.0, 2.5, 3.0, 3.4, 3.0]
+      3: Array{Float64}((150,)) [1.4, 1.4, 1.3, 1.5, 1.4, 1.7, 1.4, 1.5, 1.4, 1.5  …  5.6, 5.1, 5.1, 5.9, 5.7, 5.2, 5.0, 5.2, 5.4, 5.1]
+      4: Array{Float64}((150,)) [0.2, 0.2, 0.2, 0.2, 0.2, 0.4, 0.3, 0.2, 0.2, 0.1  …  2.4, 2.3, 1.9, 2.3, 2.5, 2.3, 1.9, 2.0, 2.3, 1.8]  Species: DataFrames.RepeatedVector{CategoricalArrays.CategoricalValue{String,UInt32}}
+    parent: CategoricalArrays.CategoricalArray{String,1,UInt32,String,Union{}}
+      refs: Array{UInt32}((150,)) UInt32[0x00000001, 0x00000001, 0x00000001, 0x00000001, 0x00000001, 0x00000001, 0x00000001, 0x00000001, 0x00000001, 0x00000001  …  0x00000003, 0x00000003, 0x00000003, 0x00000003, 0x00000003, 0x00000003, 0x00000003, 0x00000003, 0x00000003, 0x00000003]
```

Comment:
It won't pass the doctests, but if we just want to truncate the output then that's fine. To be honest I'm not sure I see the value of showing `dump` in the documentation (unless we add an advanced section to the documentation, since this is very useful but not something I'd show to a newcomer)

Reply:
Yeah, let's just remove this.